pubsub/gossipsub/partial-messages.md

```diff
@@ -0,0 +1,189 @@
+# Partial Messages Extension
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+| --------------- | ------------- | ------ | --------------- |
+| 1A              | Working Draft | Active | r0, 2025-06-23  |
+
+Authors: [@marcopolo, @cskiraly]
+
+Interest Group: TODO
+
+[@marcopolo]: https://github.com/marcopolo
+[@cskiraly]: https://github.com/cskiraly
+
+See the [lifecycle document][lifecycle-spec] for context about the maturity level
+and spec status.
+
+[lifecycle-spec]: https://github.com/libp2p/specs/blob/master/00-framework-01-spec-lifecycle.md
+
+## Overview
+
+Partial Messages Extensions allow users to transmit only a small part of a
+message rather than a full message. This is especially useful in cases where
+this a large messages and a peer is missing only a small part of the message.
+
+Much of the complexity around partial messages is contained by the Gossipsub
+implementation. Applications require little changes to benefit from this
+extension.
+
+## Terms and Definitions
+
+**Full Message**: A Gossipsub Message.
+
+**Partial Message**: Some part of a Full Message. Each part should be able to be
+validated.
+
+**Group ID**: An identifier to some Full Message. This must not depend on
+knowing the full message, so it can not simply be a hash of the full message.
+
+**Group of Partial Messages**: A set of partial messages all belonging to the
+same group. The group may be complete, meaning that all parts are available. Or
+it may be incomplete, meaning some parts are missing.
+
+## Motivation
+
+The main motivation for this extension is optimizing Ethereum's Data
+Availability Sampling (DAS) protocol. In Ethereum's upcoming fork, Fusaka,
```

Comment:
Technically this doesn't benefit the Sampling part of things. Maybe just generalize this to Ethereum's DA protocols.